tests/rust/tcp-test/accept/mod.rs

```diff
@@ -159,10 +168,18 @@ fn accept_connecting_socket(libos: &mut LibOS, remote: &SocketAddrV4) -> Result<
     // Succeed to close socket.
     libos.close(sockqd)?;
 
+    // Poll again to check that the qtoken returns an err.
+    match libos.wait(qt, Some(Duration::from_micros(0))) {
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED => {},
+        Ok(_) => anyhow::bail!("wait() should succeed with an error on connect() after close()"),
+        Err(_) => anyhow::bail!("wait() should not time out"),
+    }
+
     Ok(())
 }
 
 /// Attempts to accept connections on a TCP socket that is already accepting connections.
+/// TODO: Check if this is actually not allowed.
```

Comment:
Could you open an issue for this and link it here?

Reply:
Sure. I believe it is actually fine to call accept on a listening socket twice because accept is like pop for incoming connections.